app/controllers/cinema_halls_controller.rb

```diff
@@ -1,21 +1,26 @@
-# frozen_string_literal: true
-
 class CinemaHallsController < ApplicationController
-  # GET /cinema_halls
   def index
-    cinema_halls = CinemaHalls::UseCases::FetchAll.new.call
-    render json: CinemaHalls::Representers::All.new(cinema_halls).basic
+    cinema_halls = CinemaHalls::Repository.new.find_all
+
+    if permit_params[:extended]
+      render json: CinemaHalls::Representers::All.new(cinema_halls).extended
```

Comment:
So `extended` version of this adds `screenings`, so you would want here to do `cinema_halls.incluides(:screenings)` so those are preloaded and you are  not doing query for each and every of cinema hall